packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -734,7 +734,13 @@ class RCTAztecView: Aztec.TextView {
         case "bold": toggleBold(range: emptyRange)
         case "italic": toggleItalic(range: emptyRange)
         case "strikethrough": toggleStrikethrough(range: emptyRange)
-        case "mark": toggleMark(range: emptyRange)
+        case "mark":
+            // When there's a selection the formatting is applied from the RichText library.
+            // If not, it will toggle the active mark format if needed.
+            if selectedRange.length > 0 {
```

Comment:
Why is this same selection length check not present for the Android implementation? Conversely, Android has a no-op for `onRemoveMarkFormatting`, why is it not present here?

Reply:
The way Aztec iOS "toggles" a format is a bit different on Aztec Android.

On iOS it calls directly to each formatting style to enable it. So to prevent triggering `toggleMark` when the Highlight format was handled by the RichText component I added this check.

On Android, it just handles this by having the active styles in a HashSet ([ref](https://github.com/WordPress/gutenberg/blob/b9697d5a6941d620f8b41f847e442be710ffc278/packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecText.java#L582)) to then internally handle the different cases.